Modules/arraymodule.c

```diff
@@ -1075,10 +1075,10 @@ array_inplace_concat(arrayobject *self, PyObject *bb)
 static PyObject *
 array_inplace_repeat(arrayobject *self, Py_ssize_t n)
 {
-    char *items, *p;
-    Py_ssize_t size, i;
+    char *source;
+    Py_ssize_t size, newbytes;
```

Comment:
If declaration of `source`, `size`, and `newbytes` is touched anyway, can it be merged with their initialization? To not miss anything:
```diff
-    char *source;
-    Py_ssize_t size, newbytes;
```
```diff
-        size = Py_SIZE(self) * self->ob_descr->itemsize;
+        Py_ssize_t size = Py_SIZE(self) * self->ob_descr->itemsize;
```
```diff
-        newbytes = n * size;
-        source = self->ob_item;
+        Py_ssize_t newbytes = n * size;
+        char *source = self->ob_item;
```

It cuts off any future use of uninitialized variables in lines 1084-1092 of dense code inbetween.

Reply:
Done. I was not sure this would not interfere with caches or other optimizations.